server/task/src/main/scala/org/apache/james/task/eventsourcing/TaskAggregate.scala

```diff
@@ -30,8 +30,10 @@ import scala.jdk.CollectionConverters._
 
 class TaskAggregate private(val aggregateId: TaskAggregateId, private val history: History) {
 
-  private val currentStatus: Option[Status] = DecisionProjection
-    .load(history.getEvents.asScala.toList)
+  private val currentStatus: Option[Status] = history
+    .getEvents
+    .asScala
+    .foldLeft(Option.empty[DecisionProjection])((decision, event) => Some(decision.map(_.update(event)).getOrElse(DecisionProjection.create(event))))
```

Comment:
the code is too complex because you have that Optional[DecisionProjection].
IMO, the Status can be optional but not the DecisionProjection. If you implement an initial DecisionProjection, the code becomes much simpler.